profiles/schema.py

```diff
@@ -589,6 +598,25 @@ class Meta:
         "privileges to access this information.",
     )
 
+    def resolve_login_methods(self: Profile, info, **kwargs):
+        if info.context.user != self.user:
+            raise PermissionDenied(
+                "No permission to read login methods of another user."
+            )
+
+        amr = {info.context.user_auth.data.get("amr")}
```

Comment:
There's a small catch here. Tunnistamo returns the `amr`-claim as a string (which is not the standard way). Keycloak on the other hand will return this as a list. We could future-proof this so this doesn't break when we finally migrate over to the keycloak. e.g. [this is how it's done in linked events](https://github.com/City-of-Helsinki/linkedevents/blob/main/helevents/models.py#L36-L42).

Though once we migrate fully to keycloak we probably don't really need to check the amr-claim any more since there's no extra authentication methods available, which now exist on tunnistamo. :thinking:

Reply:
Maybe we can future-proof this and also add a note that amr-check can be removed once open-city-profile doesn't use tunnistamo anymore? :thinking:

Reply:
I actually had it as a list first, but changed it based on how it was implemented in unit tests. Will add logic for handling both cases.